src/filesystem.cpp

```diff
@@ -66,10 +66,15 @@ bool do_mkdir( std::string const &path, int const mode )
 bool assure_dir_exist( std::string const &path )
 {
     DIR *dir = opendir( path.c_str() );
+    size_t slash_idx = 0;
```

Comment:
Declare variables as lat as possible (in this case below, where an actually useful value is assigned). Declaring it here means it may not even get used at all when the function is left early via the `return true;`.

Doing this also allows to make it `const`.